comp/core/agenttelemetry/impl/agenttelemetry_test.go

```diff
@@ -113,6 +115,40 @@ func newStatusMock() statusMock {
 	return statusMock{}
 }
 
+// logger mock (currently used for trapping logged errors)
+type logMock struct {
+	loggedErr bool
+}
+
+func (l *logMock) Trace(v ...interface{})                      {} //nolint:revive
+func (l *logMock) Tracef(format string, params ...interface{}) {} //nolint:revive
+func (l *logMock) Debug(v ...interface{})                      {} //nolint:revive
+func (l *logMock) Debugf(format string, params ...interface{}) {} //nolint:revive
+func (l *logMock) Info(v ...interface{})                       {} //nolint:revive
+func (l *logMock) Infof(format string, params ...interface{})  {} //nolint:revive
+func (l *logMock) Warn(v ...interface{}) error { //nolint:revive
+	return nil
+}
+func (l *logMock) Warnf(format string, params ...interface{}) error { //nolint:revive
+	return nil
+}
+func (l *logMock) Error(v ...interface{}) error { //nolint:revive
+	l.loggedErr = true
+	return nil
+}
+func (l *logMock) Errorf(format string, params ...interface{}) error { //nolint:revive
+	l.loggedErr = true
+	return nil
+}
+func (l *logMock) Critical(v ...interface{}) error { //nolint:revive
+	return nil
+}
+func (l *logMock) Criticalf(format string, params ...interface{}) error { //nolint:revive
+	return nil
+}
+func (l *logMock) Flush() {} //nolint:revive
```

Comment:
Please use `comp/core/log/mock` and add any missing functionality there.

Reply:
Thank you. I have spent a few hours contemplating for or against it before the change. I would choose exactly what you have suggested back than but I was not sure if it would be useful to anyone. At any rate see the change in the last commit.